From 3fc5cf418a554214cbbff642fafc3e9a91aede09 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 17 Aug 2011 10:25:52 -0500
Subject: [PATCH] /etc/default/locale doesnt exist in fedora

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 rootstock |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rootstock b/rootstock
index e4ed5c7..d819023 100755
--- a/rootstock
+++ b/rootstock
@@ -691,7 +691,7 @@ BYTEINODE="4096"
 
 # default to the build system locale
 . /etc/environment
-. /etc/default/locale
+. /etc/default/locale || true
 NEWLOCALE=$LANG
 
 # default to the build system keyboard setup
-- 
1.7.4.1

