From 5d21259173c118b715773b77b752a19619899fba Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 1 Aug 2011 10:56:28 -0500
Subject: [PATCH] force unmet dependencies

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 rootstock |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/rootstock b/rootstock
index d8d7ad0..e4ed5c7 100755
--- a/rootstock
+++ b/rootstock
@@ -1149,7 +1149,9 @@ cp -f /usr/share/zoneinfo/${AREA}/${ZONE} /etc/localtime || true
 groupadd fuse || true
 
 ${APT_UPDATE}
+apt-get -f ${APT_FORCE} install
 ${APT_UPGRADE}
+apt-get -f ${APT_FORCE} install
 
 ${PACKAGE_CLEANUP}
 
@@ -1157,6 +1159,8 @@ PKGCOUNT=\$(LANG=C apt-get -s install ${SELECTION} |grep "newly installed"|cut -
 echo "packagecount=\${PKGCOUNT:-1}"
 [ -z "${SELECTION}" ] || apt-get -y ${APT_FORCE} install ${SELECTION}
 
+apt-get -f ${APT_FORCE} install
+
 groupadd admin || true
 echo '%admin  ALL=(ALL) ALL' >>/etc/sudoers
 
-- 
1.7.4.1

