=== modified file 'rootstock'
--- rootstock	2010-03-18 10:43:47 +0000
+++ rootstock	2010-04-06 14:21:25 +0000
@@ -373,9 +373,14 @@
     test -d $MOUNTPOINT/tmp/|| mkdir -p $MOUNTPOINT/tmp/
     cp $DIR/dl/$KERNEL_IMG  $MOUNTPOINT/tmp
 
-    echo "I: $KERNEL_IMG download complete"
+    [ $QUIET ] || echo "I: $KERNEL_IMG download complete"
     #Hack dpkg -i fails, no postinstall script is run anyways, so lets just extract it.
     KERNEL_IMAGE_CMD="dpkg -x /tmp/$KERNEL_IMG /"
+
+    KERNEL_IMG_REL=${KERNEL_IMG##*linux-image-}
+    KERNEL_IMG_REL=${KERNEL_IMG_REL/_*_armel.deb}
+
+    KERNEL_IMAGE_CREATE_INITRAMFS="update-initramfs -c -k $KERNEL_IMG_REL"
 }
 
 get_vm_kernel()
@@ -399,7 +404,8 @@
     dpkg -x $DIR/dl/$KERNEL_IMG $DIR/dl/tmp/
     cp $DIR/dl/tmp/boot/vmlinuz-* $DIR
     rm -rfd $DIR/dl/
-    echo "I: vmlinuz ready for (mkimage/etc) tool"
+    cp $MOUNTPOINT/boot/initrd.img-* $DIR
+    [ $QUIET ] || echo "I: vmlinuz & initrd.img ready for (mkimage/etc) tool"
 }
 
 clean_package_cache()
@@ -844,6 +850,7 @@
 [ -z "${SELECTION}" ] || apt-get -y ${APT_FORCE} install ${SELECTION}
 
 ${KERNEL_IMAGE_CMD}
+${KERNEL_IMAGE_CREATE_INITRAMFS}
 
 groupadd admin || true
 

