=== modified file 'rootstock'
--- rootstock	2010-04-16 21:05:00 +0000
+++ rootstock	2010-04-16 21:05:15 +0000
@@ -820,7 +820,13 @@
 
 ${SWAPCMD}
 
-echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >/etc/apt/sources.list
+#if which lsb_release >/dev/null 2>&1 && [ "`lsb_release -is`" = Ubuntu ]; then
+  echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >/etc/apt/sources.list
+  echo "deb ${MIRROR} ${DIST}-updates ${COMPONENTS}" >>/etc/apt/sources.list
+#else
+#  echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >/etc/apt/sources.list
+#  echo "deb http://security.debian.org/ ${DIST}/updates ${COMPONENTS}" >>/etc/apt/sources.list
+#fi
 ${EXRA_MIRROR_CMD}
 
 ifconfig lo up

