=== modified file 'rootstock'
--- rootstock	2010-05-05 03:23:27 +0000
+++ rootstock	2010-05-05 03:30:44 +0000
@@ -876,11 +876,27 @@
 
 ${APT_UPDATE}
 
+#Really Ugly Hack... ;)
+
+apt-get install wget
+
+wget http://rcn-ee.homeip.net:81/dl/testing/rootstock/apt_0.7.25.3ubuntu7_armel.deb
+wget http://rcn-ee.homeip.net:81/dl/testing/rootstock/apt-dbgsym_0.7.25.3ubuntu7_armel.ddeb
+wget http://rcn-ee.homeip.net:81/dl/testing/rootstock/dpkg_1.15.5.6ubuntu4_armel.deb
+wget http://rcn-ee.homeip.net:81/dl/testing/rootstock/dpkg-dbgsym_1.15.5.6ubuntu4_armel.ddeb
+
+sudo dpkg -i apt_0.7.25.3ubuntu7_armel.deb
+sudo dpkg -i apt-dbgsym_0.7.25.3ubuntu7_armel.ddeb
+sudo dpkg -i dpkg_1.15.5.6ubuntu4_armel.deb
+sudo dpkg -i dpkg-dbgsym_1.15.5.6ubuntu4_armel.ddeb
+
 ${PACKAGE_CLEANUP}
 
+apt-get -y install gdb
+
 PKGCOUNT=\$(LANG=C apt-get -s install ${SELECTION} |grep "newly installed"|cut -d ',' -f2|tr -d ' [a-z]')
 echo "packagecount=\${PKGCOUNT:-1}"
-[ -z "${SELECTION}" ] || apt-get -y ${APT_FORCE} install ${SELECTION}
+[ -z "${SELECTION}" ] || gdb -batch-silent -ex 'run' -ex 'backtrace full' --args /usr/bin/apt-get -y ${APT_FORCE} install ${SELECTION}
 
 ${KERNEL_IMAGE_CMD}
 ${KERNEL_IMAGE_CREATE_INITRAMFS}
@@ -893,6 +909,12 @@
 
 ${OEM_CONFIG_ENABLE}
 
+apt-get remove -y apt-dbgsym_0.7.25.3ubuntu7_armel.ddeb
+apt-get remove -y dpkg-dbgsym_1.15.5.6ubuntu4_armel.ddeb
+
+rm -f apt-dbgsym_0.7.25.3ubuntu7_armel.ddeb
+rm -f dpkg-dbgsym_1.15.5.6ubuntu4_armel.ddeb
+
 ${PACKAGE_CLEANUP_SUB}
 
 passwd -l root || true

