=== modified file 'rootstock'
--- rootstock	2010-04-14 04:09:10 +0000
+++ rootstock	2010-04-14 04:09:42 +0000
@@ -849,6 +849,19 @@
 
 ${APT_UPDATE}
 
+#Really Ugly Hack... ;)
+
+apt-get install wget
+
+#http://ddebs.ubuntu.com/pool/main/a/apt/
+wget http://ddebs.ubuntu.com/pool/main/a/apt/apt-dbgsym_0.7.25.3ubuntu6_armel.ddeb
+
+#http://ddebs.ubuntu.com/pool/main/d/dpkg/
+wget http://ddebs.ubuntu.com/pool/main/d/dpkg/dpkg-dbgsym_1.15.5.6ubuntu3_armel.ddeb
+
+sudo dpkg -i apt-dbgsym_0.7.25.3ubuntu6_armel.ddeb
+sudo dpkg -i dpkg-dbgsym_1.15.5.6ubuntu3_armel.ddeb
+
 ${PACKAGE_CLEANUP}
 
 PKGCOUNT=\$(LANG=C apt-get -s install ${SELECTION} |grep "newly installed"|cut -d ',' -f2|tr -d ' [a-z]')
@@ -866,6 +879,12 @@
 
 ${OEM_CONFIG_ENABLE}
 
+apt-get remove -y apt-dbgsym_0.7.25.3ubuntu6_armel.ddeb
+apt-get remove -y dpkg-dbgsym_1.15.5.6ubuntu3_armel.ddeb
+
+rm -f apt-dbgsym_0.7.25.3ubuntu6_armel.ddeb
+rm -f dpkg-dbgsym_1.15.5.6ubuntu3_armel.ddeb
+
 ${PACKAGE_CLEANUP_SUB}
 
 passwd -l root || true

