=== modified file 'rootstock'
--- rootstock	2010-04-16 21:05:38 +0000
+++ rootstock	2010-04-16 21:05:47 +0000
@@ -163,6 +163,8 @@
     empty the package cache of a former --copy-package-cache run
 --extra-mirror
     additional mirror to use
+--force-sec-hd <devicename>
+    use secondary hard drive in image creation    
 EOF
     exit 0
 }
@@ -189,6 +191,12 @@
     LANG=C qemu-img create $IMAGENAME $IMAGESIZE >$LOG 2>&1
     LANG=C mkfs.ext3 -F $IMAGENAME >>$LOG 2>&1
     VOLID=$(blkid -o value $IMAGENAME|head -1)
+if [ "$SEC_HD" ];then
+    [ $QUIET ] || echo "I: Formating Secondary Drive"
+    umount $DRIVENAME || true
+    LANG=c mkfs.ext3 -F $DRIVENAME >>$LOG 2>&1
+    VOLID=$(blkid -o value $DRIVENAME|head -1)
+fi
 }
 
 mount_image()
@@ -197,7 +205,12 @@
     if [ ! -d $MOUNTPOINT ];then
         mkdir -p $MOUNTPOINT >>$LOG 2>&1
     fi
-    mount -o loop $IMAGENAME $MOUNTPOINT >>$LOG 2>&1
+    if [ "$SEC_HD" ];then
+        mount $DRIVENAME $MOUNTPOINT >>$LOG 2>&1
+    else
+        mount -o loop $IMAGENAME $MOUNTPOINT >>$LOG 2>&1
+    fi
+
 }
 
 run_first_stage()
@@ -291,7 +304,13 @@
     if [ -n "$SWAPFILE" ];then
         SWAPDEV="-hdb $SWAPFILE"
     fi
+
+    if [ "$SEC_HD" ];then
+    QEMUOPTS="-M versatilepb ${VMCPU:+-cpu $VMCPU} -kernel ${BUILDDIR}/qemu-vmlinuz -no-reboot -nographic -pidfile ${QEMUPID} -hda ${DRIVENAME} ${SWAPDEV} -m 256"
+    else
     QEMUOPTS="-M versatilepb ${VMCPU:+-cpu $VMCPU} -kernel ${BUILDDIR}/qemu-vmlinuz -no-reboot -nographic -pidfile ${QEMUPID} -drive file=${IMAGENAME},aio=native,cache=none ${SWAPDEV} -m 256"
+    fi
+
     APPEND="console=ttyAMA0,115200n8 root=/dev/sda rw mem=256M devtmpfs.mount=0 init=/bin/installer quiet"
 
     qemu-system-arm $QEMUOPTS -append "${APPEND}" >$QEMUFIFO 2>&1 &
@@ -479,11 +498,14 @@
 
 save_qemu_img()
 {
+    #Not Implemented for Secondary Drive
+    if [ ! "$SEC_HD" ];then
     cp $IMAGENAME $DIR
     if [ "$SUDO_USER" ]; then
         chown $SUDO_USER $DIR/qemu-armel-$STAMP.img
     fi
     echo "I: Qemu image saved as $DIR/qemu-armel-$STAMP.img"
+    fi
 }
 
 # target system name
@@ -702,6 +724,11 @@
             checkparm $2
             EXTRAMIRROR=$2
             ;;
+        --force-sec-hd)
+            checkparm $2
+            DRIVENAME=$2
+            SEC_HD=1
+            ;;
     esac
     shift
 done

