=== modified file 'rootstock'
--- rootstock	2010-04-21 16:41:34 +0000
+++ rootstock	2010-04-21 16:43:55 +0000
@@ -823,6 +823,7 @@
 if which lsb_release >/dev/null 2>&1 && [ "`lsb_release -is`" = Ubuntu ]; then
   echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >/etc/apt/sources.list
   echo "deb ${MIRROR} ${DIST}-updates ${COMPONENTS}" >>/etc/apt/sources.list
+  touch /tmp/is-ubuntu
 else
   echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >/etc/apt/sources.list
   echo "deb http://security.debian.org/ ${DIST}/updates ${COMPONENTS}" >>/etc/apt/sources.list
@@ -833,6 +834,7 @@
 ifconfig eth0 up
 dhclient eth0
 
+if [ -x /tmp/is-ubuntu ]; then
 locale-gen --no-purge en_GB.UTF-8 || true
 locale-gen --no-purge ${NEWLOCALE} || true
 echo LANG=${NEWLOCALE} >>/etc/default/locale
@@ -841,6 +843,7 @@
 sed -i -e 's/^XKBLAYOUT=.*\$/XKBLAYOUT="${XKBL}"/' /etc/default/console-setup || true
 sed -i -e 's/^XKBVARIANT=.*\$/XKBVARIANT="${XKBV}"/' /etc/default/console-setup || true
 sed -i -e 's/^XKBOPTIONS=.*\$/XKBOPTIONS="${XKBO}"/' /etc/default/console-setup || true
+fi
 
 echo ${AREA}/${ZONE} > /etc/timezone
 cp -f /usr/share/zoneinfo/${AREA}/${ZONE} /etc/localtime || true

