=== modified file 'rootstock'
--- rootstock	2011-06-29 20:20:57 +0000
+++ rootstock	2011-06-29 20:56:16 +0000
@@ -1223,7 +1223,9 @@
 else
 	# Use system default deb source
     echo "deb ${MIRROR} ${DIST} ${COMPONENTS}" >$ROOTFS/etc/apt/sources.list
+if [ "$DIST" != "unstable" ];then
     echo "deb ${MIRROR} ${DIST}-updates ${COMPONENTS}" >>$ROOTFS/etc/apt/sources.list
+fi
 
 	# Add extra debsource
     if [ "$EXTRAMIRROR" ];then

