=== modified file 'rootstock'
--- rootstock	2011-06-14 15:51:13 +0000
+++ rootstock	2011-06-14 16:07:05 +0000
@@ -551,6 +551,7 @@
         cp $CACHEDIR/kernel/$KERNEL_IMG $ROOTFS/tmp
     else
         KERNELDIR=$(mktemp -d)
+        log "I: wget --directory-prefix=$KERNELDIR $KERNEL_URL"
         wget --directory-prefix=$KERNELDIR --quiet $KERNEL_URL
         log "I: $KERNEL_IMG download complete"
         cp $KERNELDIR/$KERNEL_IMG $ROOTFS/tmp

