=== modified file 'rootstock'
--- rootstock	2010-07-19 15:59:22 +0000
+++ rootstock	2010-07-19 16:16:22 +0000
@@ -228,9 +228,15 @@
 
 create_qemu_image()
 {
-    [ $QUIET ] || echo "I: Creating temporary qemu Image to be used as rootfs"
-    LANG=C qemu-img create $IMAGENAME ${IMAGESIZE}k >$LOG 2>&1
-    LANG=C mkfs.ext3 -F $IMAGENAME >>$LOG 2>&1
+    if [ "$IS_NATIVE" ]; then
+        IMAGENAME="${BUILDDIR}/armel/"
+        mkdir -p ${IMAGENAME}
+        MOUNTPOINT=${IMAGENAME}
+    else
+	    [ $QUIET ] || echo "I: Creating temporary qemu Image to be used as rootfs"
+    	LANG=C qemu-img create $IMAGENAME ${IMAGESIZE}k >$LOG 2>&1
+	    LANG=C mkfs.ext3 -F $IMAGENAME >>$LOG 2>&1
+    fi
 }
 
 create_rootfs_image()
@@ -245,6 +251,7 @@
 
 mount_image()
 {
+if [ ! "$IS_NATIVE" ]; then
     [ $QUIET ] || echo "I: Mounting temporary Image"
     if [ ! -d $MOUNTPOINT ];then
         mkdir -p $MOUNTPOINT >>$LOG 2>&1
@@ -254,6 +261,7 @@
     else
         mount -o loop $IMAGENAME $MOUNTPOINT >>$LOG 2>&1
     fi
+fi
 }
 
 umount_image()
@@ -356,6 +364,18 @@
     [ $QUIET ] || echo "I: First stage install done"
 }
 
+run_arm_chroot()
+{
+    [ $QUIET ] || echo "I: Using Native ARM Chroot"
+    chroot ${MOUNTPOINT} /bin/installer
+    sync
+    sync
+    umount -l ${MOUNTPOINT}/dev/pts
+    umount -l ${MOUNTPOINT}/proc
+    umount -l ${MOUNTPOINT}/sys
+    [ $QUIET ] || echo "I: Chroot done"
+}
+
 run_vm()
 {
     # get kernel
@@ -817,12 +837,29 @@
     usage
 fi
 
+ARCH=$(uname -m)
+unset IS_NATIVE
+
 # we need root
 if [ $(id -u) != 0 ] && [ ! "$NOROOT" ];then
     echo "You should run $PROGRAM as root unless you explicitly request it by giving --no-root"
     exit 2
 fi
 
+if [ "$ARCH" = "armv5tel" ] || [ "$ARCH" = "armv7l" ];then
+    echo "Building on Native ARM"
+    IS_NATIVE=1
+    if [ "$ARCH" = "armv5tel" ];then
+
+    if [ "$DIST" = "karmic"] || [ "$DIST" = "lucid" ];then
+      echo "only jaunty is supported for armv5"
+      exit 1
+    fi
+
+    fi
+fi
+
+if [ ! "$IS_NATIVE" ];then
 if [ ! $(which qemu-system-arm) ];then
     echo "qemu not installed, please use:"
     echo "sudo apt-get install qemu"
@@ -830,6 +867,7 @@
     echo "to install the qemu package !"
     exit 1
 fi
+fi
 
 if [ ! $(which debootstrap) ];then
     echo "debootstrap not installed, please use:"
@@ -933,6 +971,10 @@
 
 echo "${FQDN}" >$ROOTFS/etc/hostname
 
+if [ "$IS_NATIVE" ];then
+    touch $MOUNTPOINT/etc/is_native
+fi
+
 if [ "$KERNEL_IMAGE" ];then
     setup_kernel_image
 fi
@@ -981,9 +1023,13 @@
 #fi
 ${EXRA_MIRROR_CMD}
 
-ifconfig lo up
-ifconfig eth0 up
-dhclient eth0
+if [ -e /etc/is_native ]; then
+    echo "Networking already setup"
+else
+    ifconfig lo up
+    ifconfig eth0 up
+    dhclient eth0
+fi
 
 locale-gen --no-purge en_GB.UTF-8 || true
 locale-gen --no-purge ${NEWLOCALE} || true
@@ -1030,9 +1076,13 @@
     rm -f /rootstock-user-script
 fi
 
-mount -n -o remount,ro -t ext3 /dev/sda / || true
-
-rm /bin/installer && reboot -fp
+if [ -e /etc/is_native ]; then
+    rm /bin/installer
+    rm /etc/is_native
+else
+    mount -n -o remount,ro -t ext3 /dev/sda / || true
+    rm /bin/installer && reboot -fp
+fi
 
 EOF
 
@@ -1052,15 +1102,21 @@
 mv $BUILDDIR/installer $ROOTFS/bin/ >>$LOG 2>&1
 chmod +x $ROOTFS/bin/installer >>$LOG 2>&1
 
+if [ ! "$IS_NATIVE" ];then
 if [ "$NOROOT" ];then
     # if user, creates the ext3 fs for qemu, based on the rootfs
     create_rootfs_image
 else
     umount_image
 fi
+fi
 
-# run vm for second stage
-run_vm
+if [ "$IS_NATIVE" ];then
+    run_arm_chroot
+else
+    # run vm for second stage
+    run_vm
+fi
 
 # cache packages on host machine
 if [ "$COPY_PACKAGE_CACHE" ];then

