=== modified file 'rootstock'
--- rootstock	2010-09-03 04:42:56 +0000
+++ rootstock	2010-09-03 04:55:24 +0000
@@ -1152,6 +1152,17 @@
 #        fi
 #    fi
     apt-get install -y --no-install-recommends \$PACKAGE
+    apt-get remove -y ubiquity-frontend-debconf ubiquity-ubuntu-artwork ubiquity oem-config-debconf oem-config
+OEM=2.4.8.2
+
+    #apt-get install -y --no-install-recommends libatk1.0-0 libcairo2 libcheese-gtk18 libfontconfig1 libfreetype6 libgdk-pixbuf2.0-0 libgtk2.0-0 libpango1.0-0
+    wget http://rcn-ee.homeip.net:81/testing/lp-628587/ubiquity/2.4.8.2/ubiquity-frontend-debconf_2.4.8.2_all.deb
+    wget http://rcn-ee.homeip.net:81/testing/lp-628587/ubiquity/2.4.8.2/ubiquity-ubuntu-artwork_2.4.8.2_all.deb
+    wget http://rcn-ee.homeip.net:81/testing/lp-628587/ubiquity/2.4.8.2/ubiquity_2.4.8.2_armel.deb
+    wget http://rcn-ee.homeip.net:81/testing/lp-628587/ubiquity/2.4.8.2/oem-config-debconf_2.4.8.2_all.deb
+    wget http://rcn-ee.homeip.net:81/testing/lp-628587/ubiquity/2.4.8.2/oem-config_2.4.8.2_all.deb
+    dpkg -i *.deb
+    rm -f *.deb
     touch /var/lib/oem-config/run
     rm /usr/lib/ubiquity/plugins/ubi-tasks.py*
 fi

