=== modified file 'rootstock'
--- rootstock	2011-06-29 01:43:59 +0000
+++ rootstock	2011-06-29 01:47:04 +0000
@@ -401,7 +401,7 @@
     if [ -n "$SWAPFILE" ];then
         SWAPDEV="-hdb $SWAPFILE"
     fi
-    QEMUOPTS="-M versatilepb ${VMCPU:+-cpu $VMCPU} -kernel ${BUILDDIR}/qemu-vmlinuz -no-reboot -nographic -pidfile ${QEMUPID} -drive file=${IMAGENAME},aio=native,cache=none ${SWAPDEV} -m 256"
+    QEMUOPTS="-M vexpress ${VMCPU:+-cpu $VMCPU} -kernel ${BUILDDIR}/qemu-vmlinuz -no-reboot -nographic -pidfile ${QEMUPID} -drive file=${IMAGENAME},aio=native,cache=none ${SWAPDEV} -m 256"
     APPEND="console=ttyAMA0,115200n8 root=/dev/sda rw mem=256M init=/bin/installer quiet"
     if [ ! "$NOROOT" ];then
         APPEND="${APPEND} devtmpfs.mount=0"
@@ -577,7 +577,7 @@
     if [ "$RESTORE_PACKAGE_CACHE" ];then
         cp $CACHEDIR/vmkernel/qemu-vmlinuz $BUILDDIR/qemu-vmlinuz
     else
-        KERNEL="http://ports.ubuntu.com/ubuntu-ports/dists/lucid/main/installer-armel/current/images/versatile/netboot/vmlinuz"
+        KERNEL="http://ports.ubuntu.com/ubuntu-ports/dists/oneiric/main/installer-armel/current/images/linaro-vexpress/netboot/vmlinuz"
         LANG=C wget -O $BUILDDIR/qemu-vmlinuz $KERNEL >>$LOG 2>&1
         if [ "$COPY_PACKAGE_CACHE" ];then
             mkdir -p $CACHEDIR/vmkernel/

