=== modified file 'rootstock'
--- rootstock	2011-07-06 16:54:42 +0000
+++ rootstock	2011-07-06 16:57:34 +0000
@@ -570,6 +570,7 @@
     KERNEL_IMG_REL=$(echo $KERNEL_IMG_REL | awk -F_ '{print $1}')
 
     KERNEL_IMAGE_CREATE_INITRAMFS="update-initramfs -c -k $KERNEL_IMG_REL"
+    KERNEL_IMAGE_DEPMOD="depmod $KERNEL_IMG_REL"
 }
 
 get_vm_kernel()
@@ -1187,6 +1188,8 @@
 ${KERNEL_IMAGE_CMD}
 if which update-initramfs >/dev/null 2>&1; then
 ${KERNEL_IMAGE_CREATE_INITRAMFS}
+else
+${KERNEL_IMAGE_DEPMOD}
 fi
 
 ${PACKAGE_CLEANUP_SUB}

